### PR TITLE
[SPARK-7554][STREAMING] Throw exception when an active/stopped StreamingContext is used to create DStreams and output operations

### PR DESCRIPTION
**Pull request by tdas**

### CODE REVIEW

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala [hunk 1]
```diff
@@ -171,7 +173,22 @@ abstract class DStream[T: ClassTag] (
     dependencies.foreach(_.initialize(zeroTime))
   }
 
-  private[streaming] def validate() {
+  private def validateAtInit(): Unit = {
```
Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala]: I renamed `validate()` to `validateAtStart()`, and added `validateAtInit()`

**Comment by tdas** [comment]
@zsxwing

**Comment by zsxwing** [comment]
LGTM

**Comment by tdas** [comment]
@harishreedharan Can you take a look?

**Comment by harishreedharan** [comment]
+1

**Comment by tdas** [comment]
Merging this. Thanks all!

**Pull request closed without merging**